app/src/main/java/com/mk/tasky/agenda/presentation/detail/task/DetailTaskViewModel.kt

```diff
@@ -5,13 +5,20 @@ import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.setValue
 import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import com.mk.tasky.agenda.domain.usecase.task.TaskUseCases
+import com.mk.tasky.agenda.presentation.detail.components.model.NotificationTypes
 import com.mk.tasky.agenda.presentation.home.HomeItemOptions
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.NonCancellable
+import kotlinx.coroutines.launch
 import java.time.LocalDateTime
 import javax.inject.Inject
 
+@HiltViewModel
 class DetailTaskViewModel @Inject constructor(
-    savedStateHandle: SavedStateHandle
+    savedStateHandle: SavedStateHandle,
+    private val taskUseCases: TaskUseCases
 ) : ViewModel() {
     var state by mutableStateOf(DetailTaskState())
         private set
```

Comment:
BTW, not sure if you want to do that already or do it at the end, but right now your app wouldn't survive process death

Reply:
I'm going to do it at the end of the whole app, as the "final touch"